src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -0,0 +1,400 @@
+    } else {
+        config.emplace("NPUW_FUNCALL_FOR_ALL", "YES");
+    }
+    return config;
+}
+
+ov::AnyMap get_default_prefill_config(const std::shared_ptr<ov::Model>& model,
+                                      const std::optional<NPUDesc>& npudesc) {
+    auto config = get_default_common_config(model);
+    if (is_cw_compressed(model)) {
+        config.emplace("NPUW_DQ", "YES");
+    } else {
+        config.emplace("NPUW_PMM", "NO");
+    }
+    if (npudesc.has_value() &&
+        npudesc->arch == "4000" &&
+        npudesc->max_tiles != -1) {
+        config.emplace("NPU_DPU_GROUPS", npudesc->max_tiles);
+    }
+    return config;
+}
+
+ov::AnyMap get_default_generate_config(const std::shared_ptr<ov::Model>& model,
+                                       const std::optional<NPUDesc>& npudesc,
+                                       const GenerateHint hint) {
+    auto config = get_default_common_config(model);
+    if (hint == GenerateHint::BEST_PERF) {
+        config.emplace("NPUW_ONLINE_PIPELINE", "NONE");
+    }
+    // NB: Unconditionally set for generation model
+    config.emplace("NPUW_DQ", "YES");
+    if (npudesc.has_value() && npudesc->arch == "4000") {
+        config.emplace("NPU_DPU_GROUPS", 4);
+    }
+    return config;
+}
+
+void merge_config_with(ov::AnyMap& lhs, const ov::AnyMap& rhs) {
+    for (const auto& [key, value] : rhs) {
+        // NB: Overwrite the value if key already exists
+        if (auto it = lhs.find(key); it != lhs.end()) {
+            it->second = value;
+        } else {
+            lhs.emplace(key, value);
+        }
+    }
+}
+
+void drop_cache_dir(ov::AnyMap& config) {
+    if (config.count("NPU_USE_NPUW") != 0u) {
+        pop_option(config, "CACHE_DIR");
+    }
+}
+
+void split_properties(const ov::AnyMap& properties,
+                      ov::AnyMap& dyn_llm_properties,
+                      ov::AnyMap& other_properties) {
+    for (auto it = properties.begin(); it != properties.end(); ++it) {
+        if (it->first.find("NPUW_DYN_LLM") != it->first.npos) {
+            dyn_llm_properties.insert(*it);
+        } else {
+            other_properties.insert(*it);
+        }
+    }
+}
+
+std::map<std::string, std::string> any_copy(const ov::AnyMap& params) {
+    std::map<std::string, std::string> result;
+    for (auto&& value : params) {
+        result.emplace(value.first, value.second.as<std::string>());
+    }
+    return result;
+}
+} // namespace
+
+ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& model,
+                                             const std::shared_ptr<const ov::IPlugin>& plugin,
+                                             const ov::AnyMap& properties)
+    : ov::npuw::ICompiledModel(model, plugin),
+     m_options_desc(std::make_shared<::intel_npu::OptionsDesc>()),
+     m_cfg(m_options_desc) {
+    LOG_DEBUG("LLMCompiledModel::LLMCompiledModel");
+    LOG_BLOCK();
+
+    ::intel_npu::registerNPUWOptions(*m_options_desc);
```

Comment:
Should it be called here?
Are all NPUW options routed through this mechanism?

Reply:
Thank you!
I have added this call here to register new options `NPUW_LLM*` in config for `LLMCompiledModel`. However, here are really all options of `NPUW*` will be registered, fixed it.

Reply:
Did you?

Reply:
Sorry, haven't uploaded last changes! Now it should be fixed!